src/__test__/extra/write-after-end.test.ts

```diff
@@ -94,12 +94,7 @@ describe("write after end", () => {
 
       const error = await errorPromise;
 
-      // The reason for this discrepancy is explained in mid-stream.test.ts
-      if (matchServerVersion`<=23.10`) {
-        expect(error).toBeInstanceOf(CancelledError);
-      } else {
-        expect(error).toBeInstanceOf(UnavailableError);
-      }
+      expect(error).toBeInstanceOf(CancelledError);
```

Comment:
Seems to still be consistently failing on next-feature-version:
![image](https://github.com/EventStore/EventStore-Client-NodeJS/assets/11861797/33ccc711-a1c3-4002-8232-40218b081835)

Reply:
It seems the error in the screenshot (line 31) does not match the original codebase (line 97). This behavior is new and unrelated to the previous release. It seems to be a timing sensitivity issue. I introduced some delay when killing a node.